crates/core_simd/src/lib.rs

```diff
@@ -14,6 +14,9 @@ mod intrinsics;
 mod ops;
 mod round;
 
+mod lanes_at_most_64;
+pub use lanes_at_most_64::*;
```

Comment:
Do we need to actually publicly export this trait?

I guess ideally what we want is to be able to bound the const generic when that becomes possible.

If users do need to use this trait (such as in generic code) then maybe we could come up with a different name for it and seal it so that it's a bit less of a visible hack?

What do you think?

Reply:
it needs to be visible for generics

name bikeshed begins

Reply:
If it needs to be visible and possibly stabilized maybe we could possibly call it something like `FixedSimd` and turn it into a useful trait?

```rust
#[rustc_on_unimplemented="fixed SIMD vectors are implemented for types with lane sizes of 2, 4, 8, 16, 32, or 64"]
pub trait FixedSimd {
    type T;
    const N: usize;
}
```

Reply:
I don't think it needs to live into stable. At least I hope it doesn't.

But for now, on nightly, users do need to see it.

Reply:
I agree that it doesn't need to be stable.  You can do quite a bit with generics without touching the trait: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=46c81ab903379f78b15b03ec98eab68e

Reply:
If we don't want to stabilize it then any name seems ok to me 🙂 We can at least use the `private::Sealed` trick to prevent anybody else from implementing it.

Reply:
Do we need to use a glob import here? Seems like it risks exposing something in the public API unintentionally?

Reply:
No significant risk.

When adding items you don't want to leave the crate you should be marking them as `pub(crate)` anyway.

Reply:
Doesn't mean it will happen. Doesn't seem worth the risk to me if you're relying on folks using `pub(crate)`. I certainly don't always do that. The glob import doesn't seem worth it to me.

Reply:
For this particular glob, yes it's trivial to turn it into a plain import.  The other globs import quite a bit and we rely on using `pub` and `pub(crate)` to not forget anything.  I think it could go either way.